### ISSUE TITLE
`D102` for "inherited" docstrings.

### ISSUE THREAD

**Issue by WilliamJamieson** [charliermarsh/ruff#2413]
`pydocstyle` code `D102` is raised on any public methods that do not have a docstring. I agree with this in principle, except that I have many cases where I have a common base class which defines some method's interface (and documentation), but that method's specific implementation will be defined by child classes or needs to be modified by a child class. E.G.

```python
class Parent:
    def foo(self, arg1, arg2, arg3):
        """Some documentation about foo."""


# possibly in a different module or package
class Child(Parent):
    def foo(self, arg1, arg2, arg3):
		return arg1 + arg2 + arg3
```

`ruff` flags `foo` in `Child` with `D102`, when in fact this is documented by `Parent.foo`. 

In most cases, documenting the method in the child class would be equivalent to copy/paste of the docstring from the parent class, which I dislike. I would like the optional ability for `ruff` to ignore `D102` on methods defined in their parent classes.

**Comment by charliermarsh** [comment]
I agree that this would be an improvement. We probably can't really support it until we start doing cross-file analysis (otherwise, it would only work if `Parent` and `Child` were defined in the same module, which seems like a minority of cases).

**Comment by norabelrose** [comment]
Would just like to mention that I would also really like to see this feature, since this is the main reason I'm not using the `D` rules yet. Unfortunately I'm not proficient enough in Rust to contribute to implementing it 😅

**Comment by fjwillemsen** [comment]
It would be a nice start to ignore it in the same file, can that be implemented on short notice?

**Comment by philipp-horstenkamp** [comment]
This issue is the only drawback I am having with ruff.
Everything else is better. But this is a huge problem for me.

**Comment by davfsa** [comment]
This is a bit of a bump, but I wanted to ask if cross-file analysis is now a more supported thing in ruff (as you pointed out @charliermarsh), in which case I would be willing to give implementing this a shot :)

Its a bit of a blocker for one of our projects, and I have resorted to just disabling `D102` entirely for the time being (but would love to active it at some point!)

**Comment by MichaReiser** [comment]
@davfsa no, multifile analysis is still in progress.

**Comment by tahaum** [comment]
Would be nice! 👍

**Comment by Avasam** [comment]
This can be supported without mukti-fike analysis with the [`@override`](https://typing.python.org/en/latest/spec/class-compat.html#override) decorator.

**Comment by analog-cbarber** [comment]
The `@override` decorator does not tell you whether the parent method has a doc string, so even with that you cannot tell.

**Comment by Avasam** [comment]
> The `@override` decorator does not tell you whether the parent method has a doc string, so even with that you cannot tell.

Oh I guess you'd still want to trigger when no base method has a docstring.

Depends if you'd prefer false-positives or false-negatives here. But not as straightforward as I thought.

**Comment by philipp-horstenkamp** [comment]
This would need to be configurable.

Am 14. Mai 2025 15:47:50 MESZ schrieb Avasam ***@***.***>:
